### PR TITLE
feat(slice-2b): frontend infra — date-derived current week, toaster mount, FormMessage alert role

### PR DESCRIPTION
**Pull request by leehopper**
## PR6a — Frontend infra (slice 2b workout logging)

First of the frontend PRs in the stacked slice-2b train (after **PR1 #134**). Pre-stages the `/log` form infra and ships the always-week-1 fix early. Depends only on PR1 (the `PlanStartDate` anchor), already on `main`.

### What changed
- **`resolveCurrentWeek` is now date-derived** from the plan's `PlanStartDate` anchor (DEC-076 / Unit 1) relative to today: `floor((today − planStartDate).days / 7) + 1`, clamped into the meso-template week range so the home surface always lands on a renderable week. This replaces the lowest-populated-week heuristic that pinned the home page to week 1; the heuristic is retained as a defensive fallback for streams with no usable anchor. The reference date is injectable for deterministic tests.
- **Mounted a single app-wide `<Toaster>`** (sonner) as a sibling outside the error boundary, so slice-2b's log-success toast (PR6b) can render app-wide. The `sonner.tsx` primitive shipped in 2a (#111) but was never mounted.
- **`FormMessage` gains `role="alert"`** for assertive screen-reader error announcement (carry-in #560) — matching the hand-rolled onboarding inputs. The role is scoped to actual field errors only (not static children rendered through the slot), so the ARIA semantics stay correct.

### Also in this push (opportunistic test hygiene)
- **Silenced the `[FAIL] check-contrast` console noise during `npm run test`.** Not a real contrast failure — the gate's `exits non-zero when a token regresses` integration test deliberately breaks `--ctp-text` and spawns the gate via `execFileSync`, whose default stderr was inherited to the console. Piping stderr (`stdio: ['ignore','pipe','pipe']`) keeps the existing `error.stderr` assertions working while stopping the leak. Test-only change.

### Tests / gates
- **+8** `resolveCurrentWeek` unit specs: in-range derivation (no lowest-populated fallback), first-day → week 1, mid-week floor division, in-range week with no micro detail, pre-start clamp, beyond-generated-weeks clamp, malformed-anchor fallback.
- **+3** `FormMessage` specs: `role="alert"` on error; no alert when clean; static children render without the role.
- **+1** `<Toaster>` mount test: a fired toast renders app-wide.
- Frontend **379/379** Vitest green (367 → +12); `tsc -b` + `vite build` clean; `eslint` 0 errors.

### Note
`PlanStartDate` was already mirrored onto the hand-maintained `plan.model.ts` in PR1, so no model change was needed here — narrowed from the PR-strategy description, which predated PR1's deep-review widening.

### Follow-ups found
None.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added app-wide toast notifications for user feedback.
  * Implemented date-based current week calculation for workout plans.

* **Improvements**
  * Enhanced form message accessibility with screen-reader alerts for validation errors.

* **Tests**
  * Added comprehensive test coverage for toast notifications and error handling.
  * Expanded tests for week derivation logic and edge cases.
  * Added tests for form message component behavior.

### CODE REVIEW

File: frontend/src/app/modules/plan/hooks/use-plan.hooks.ts [hunk 1]
```diff
@@ -58,12 +71,52 @@ export const resolveCurrentWeek = (plan: PlanProjectionDto): number => {
...
-  // Fall back to the first meso template's week number when no micro
-  // workouts are present (defensive path for a partially-projected stream).
   const firstMeso = plan.mesoWeeks.find((week) => week.weekNumber >= 1)
   return firstMeso?.weekNumber ?? 1
 }
 
+/**
+ * Returns the 1-based current training week, derived from the plan's calendar
+ * anchor (`planStartDate`, DEC-076 / slice-2b Unit 1) relative to `referenceDate`
+ * (today by default): `floor((referenceDate − planStartDate).days / 7) + 1`.
```
Comment (leehopper) [inline frontend/src/app/modules/plan/hooks/use-plan.hooks.ts:81]: **💡 [LOW] resolveCurrentWeek doc comment misattributes the calendar anchor to DEC-076**

**💡 [low] [documentation] resolveCurrentWeek JSDoc misattributes the calendar anchor to DEC-076**

The `resolveCurrentWeek` JSDoc cites **DEC-076** as the authority for the `planStartDate` calendar anchor, but DEC-076 in the decision log is *“WorkoutLog ↔ prescribed-workout linkage — snapshot-on-log, not a foreign key”* — it governs the WorkoutLog snapshot shape, not the calendar anchor or current-week derivation. No decision-log entry covers the anchor (confirmed by the conventions, validator, and blind-challenge agents), so the citation points a reader to the wrong record. The “slice-2b Unit 1” half is accurate.

**Suggested fix:** drop the `DEC-076` citation (keep `slice-2b Unit 1`), or add a real DEC for the calendar anchor and cite that.

---
*Confidence: 75% | deep-review*

```suggestion
 * anchor (`planStartDate`, slice-2b Unit 1) relative to `referenceDate`
 * (today by default): `floor((referenceDate − planStartDate).days / 7) + 1`.
 *
```

File: frontend/src/app/modules/plan/hooks/use-plan.hooks.ts [hunk 2]
```diff
@@ -58,12 +71,52 @@ export const resolveCurrentWeek = (plan: PlanProjectionDto): number => {
...
+    .map((week) => week.weekNumber)
+    .filter((value) => Number.isFinite(value) && value >= 1)
+    .sort((left, right) => left - right)
+
+  const planStartUtc = parsePlanStartUtc(plan.planStartDate)
+  const firstWeek = templateWeeks[0]
+  const lastWeek = templateWeeks[templateWeeks.length - 1]
+  if (planStartUtc === undefined || firstWeek === undefined || lastWeek === undefined) {
+    return resolveLowestPopulatedWeek(plan)
+  }
```
Comment (leehopper) [inline frontend/src/app/modules/plan/hooks/use-plan.hooks.ts:107]: **💡 [LOW] Fallback path for empty mesoWeeks / fully-unpopulated plan in resolveCurrentWeek is untested**

**💡 [low] [test coverage] Fallback path for empty `mesoWeeks` / fully-unpopulated plan is untested**

`resolveCurrentWeek` falls back to `resolveLowestPopulatedWeek` when `firstWeek`/`lastWeek` is undefined (empty `mesoWeeks`) or `planStartDate` is unusable. The malformed-date spec only exercises the `firstPopulated !== undefined` early return (returns 2). The deeper branches — empty `mesoWeeks` → `firstMeso?.weekNumber`, and both maps empty → the literal `?? 1` guard — are untested. Since `resolveLowestPopulatedWeek` is now private, these are reachable only through `resolveCurrentWeek`, and the home page renders directly off this value; a regression dropping the `?? 1` guard would ship undetected and break week selection on a partially-projected stream.

**Suggested fix:** add specs for (a) a valid `planStartDate` with `mesoWeeks: []`, and (b) both `mesoWeeks` and `microWorkoutsByWeek` empty → asserts `1`.

---
*Confidence: 70% | deep-review*

File: frontend/src/app/modules/app/app.component.tsx [hunk 3]
```diff
@@ -189,6 +190,11 @@ export const App = () => {
           <AppShell />
         </BrowserRouter>
       </AppErrorBoundary>
+      {/* Single app-wide toast outlet. Mounted as a sibling *outside* the
+          error boundary so a render-time throw in the route tree cannot
+          unmount it, and toasts (e.g. the slice-2b log success) still surface
+          on the recovery card. `Toaster` mirrors the `.dark` class itself. */}
+      <Toaster />
```
Comment (leehopper) [inline frontend/src/app/modules/app/app.component.tsx:197]: **💡 [LOW] Toast test does not verify the load-bearing placement outside the error boundary**

**💡 [low] [test coverage] Toast test does not verify the load-bearing placement outside the error boundary**

The `<Toaster>` is deliberately mounted as a sibling *outside* `AppErrorBoundary` so toasts survive a render-time throw and still surface on the recovery card. The new spec only fires a toast under the normal loading state — no test renders a throwing route to make the boundary catch and then re-asserts a toast still surfaces. A refactor moving `<Toaster>` back inside the boundary would pass all tests while silently breaking the slice-2b log-success toast during error recovery.

**Suggested fix:** add a spec that renders a throwing route, confirms the recovery card, then fires `toast.success` and asserts the toast text still appears.

---
*Confidence: 65% | deep-review*

**Review by leehopper — COMMENTED** [review]
### Deep Review

Reviewed PR #136 in **Frontier mode** (7 Opus discovery agents + Sonnet validation + Opus blind-challenge verification). **No blocking issues** — the change is clean on correctness, security, cross-file impact, type design, and code simplification.

**3 low-severity improvement suggestions** are posted inline: 2 test-coverage gaps (`resolveCurrentWeek` fallback branches; the `<Toaster>` survives-error-boundary guarantee) and 1 documentation-citation fix (`DEC-076` misattribution). All three were independently confirmed by the blind-challenge round (scores 96–97).

---
Generated by deep-review | Reviewed up to: 8e421f1008c280ea8f144aff7fc2196a6b63875c

File: frontend/src/app/modules/plan/hooks/use-plan.hooks.ts (hunk 1 above)
Comment (leehopper) [reply frontend/src/app/modules/plan/hooks/use-plan.hooks.ts:81]: Fixed in 449ae2e — dropped the `DEC-076` citation from the `resolveCurrentWeek` JSDoc (and the same misattribution in the spec header comment), keeping the accurate `slice-2b Unit 1` reference.

File: frontend/src/app/modules/plan/hooks/use-plan.hooks.ts (hunk 2 above)
Comment (leehopper) [reply frontend/src/app/modules/plan/hooks/use-plan.hooks.ts:107]: Fixed in 449ae2e — added three `resolveCurrentWeek` specs covering the previously-untested `resolveLowestPopulatedWeek` fallback branches: empty `mesoWeeks` → lowest populated micro week (3), malformed anchor + empty micro → first meso template week (1), and neither templates nor micro → the `?? 1` guard (1).

File: frontend/src/app/modules/app/app.component.tsx (hunk 3 above)
Comment (leehopper) [reply frontend/src/app/modules/app/app.component.tsx:197]: Fixed in 449ae2e — added an app spec that forces a render-time throw inside `AppErrorBoundary` and asserts the out-of-boundary `<Toaster>` survives the catch and still surfaces a toast. Used a persistent throw mock (not `mockImplementationOnce`) so React 19 routes the throw to the boundary across both the concurrent and synchronous-recovery render passes.
